src/psyclone/dynamo0p3.py

```diff
@@ -865,26 +892,28 @@ def __init__(self, ast, name=None):
         # set to True if all the checks in _validate_inter_grid() pass.
         self._is_intergrid = False
 
+        # The properties of the Reference Element that this kernel
```

Comment:
The code looks good, however I think it would be good to move the implementation of reference element metadata into a separate object or class instead of further increasing this class `__init__` method. I think many Dynamo0.3 classes have overblown `__init__` methods that are doing way more than a simple initialisation - complex parsing of metadata, checking and setting properties could be done in a separate private method that a class inititialiser could then refer to.